test/dynamicppl/compiler.jl

```diff
@@ -0,0 +1,370 @@
+module DynamicPPLCompilerTests
+
+using ..NumericalTests: check_numerical
+using LinearAlgebra: I
+using Test: @test, @testset, @test_throws
+using Turing
+
+# TODO(penelopeysm): Move this to a DynamicPPL Test Utils module
```

Comment:
This is equivalent to one of the DynamicPPL.TestUtils models; I believe it's the same as `demo_assume_literal_observe`?

Reply:
It's the same as this one https://github.com/TuringLang/DynamicPPL.jl/blob/ba6e3b8d5337b36b3ea680fa3de393d05bd85d34/src/test_utils/models.jl#L145-L152 but we don't actually have a copy in the demo models

Reply:
Ah, okay, but then at least we could use that / make that parto f demo models

Reply:
Doing this as part of https://github.com/TuringLang/DynamicPPL.jl/pull/733